operator/builtin/parser/csv/csv_test.go

```diff
@@ -0,0 +1,232 @@
+package csv
+
+import (
+	"context"
+	"testing"
+
+	"github.com/observiq/stanza/entry"
+	"github.com/observiq/stanza/operator"
+	"github.com/observiq/stanza/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+var testHeader = "name,sev,msg"
+
+func newTestParser(t *testing.T) *CSVParser {
+	cfg := NewCSVParserConfig("test")
+	cfg.Header = testHeader
+	ops, err := cfg.Build(testutil.NewBuildContext(t))
+	require.NoError(t, err)
+	op := ops[0]
+	return op.(*CSVParser)
+}
+
+func TestCSVParserBuildFailure(t *testing.T) {
+	cfg := NewCSVParserConfig("test")
+	cfg.OnError = "invalid_on_error"
+	_, err := cfg.Build(testutil.NewBuildContext(t))
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "invalid `on_error` field")
+}
+
+func TestCSVParserStringFailure(t *testing.T) {
+	parser := newTestParser(t)
+	_, err := parser.parse("invalid")
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "record on line 1: wrong number of fields")
+}
+
+func TestCSVParserByteFailure(t *testing.T) {
+  parser := newTestParser(t)
+	_, err := parser.parse([]byte("invalid"))
+	require.Error(t, err)
+  require.Contains(t, err.Error(), "record on line 1: wrong number of fields")
+}
+
+func TestCSVParserInvalidType(t *testing.T) {
+  parser := newTestParser(t)
+	_, err := parser.parse([]int{})
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "type '[]int' cannot be parsed as csv")
+}
+
```

Comment:
We should test some additional invalid inputs including:
- unparsed json
- wrong delimiter
- multiple lines (especially where they are each valid on their own)